lib/watir_drops/page_object.rb

```diff
@@ -97,8 +115,27 @@ def fill_form(model)
     def inspect
       '#<%s url=%s title=%s>' % [self.class, url.inspect, title.inspect]
     end
+
     alias selector_string inspect
 
+
+    def on_page?
+      if self.respond_to?('page_url')
+        Watir::Wait.until { page_url.gsub(/.*:\/\//i,'').gsub(/\/$/i,'') == (@browser.url.gsub(/.*:\/\//i,'').gsub(/\/$/i,'')) }
```

Comment:
stripping out any https:// or http:// that might differ between how the user defined their page_url and what the browser returns. I know different browsers return different values for URL at times. I imagine there is a better way to do this. 

However, we could also be more aggressive and require the user ensure their page_url matches exactly what they have in their page object (meaning you should use the URL as it exists after any redirects / manipulations)